pydantic/main.py

```diff
@@ -753,7 +755,7 @@ def _get_value(
                 and (not value_include or value_include.is_included(i))
             )
 
-        elif isinstance(v, Enum) and getattr(cls.Config, 'use_enum_values', False):
+        elif (value_type == Enum or isinstance(v, Enum)) and getattr(cls.Config, 'use_enum_values', False):
```

Comment:
Again I think here Enum is almost always subclassed?

Reply:
I was thinking since we already grabbed the type from the value. Then wouldn't it be faster to use that value? Couldn't we go as far as to cut out `isinstance` entirely?

Reply:
I just think `value_type == Enum` will never be true because you always subclass it (see example in https://docs.python.org/3/library/enum.html)